src/java/org/apache/cassandra/db/CassandraKeyspaceWriteHandler.java

```diff
@@ -61,6 +64,43 @@ public WriteContext beginWrite(Mutation mutation, boolean makeDurable) throws Re
         }
     }
 
+    private CommitLogPosition addToCommitLog(Mutation mutation)
+    {
+        CommitLogPosition position;
+        // Usually one of these will be true, so first check if that's the case.
+        boolean allSkipCommitlog = true;
+        boolean noneSkipCommitlog = true;
+        for (TableId id : mutation.getTableIds())
+        {
+            if (keyspace.getColumnFamilyStore(id).writesShouldSkipCommitLog())
+                noneSkipCommitlog = false;
+            else
+                allSkipCommitlog = false;
+        }
+
+        if (!noneSkipCommitlog)
+        {
+            if (allSkipCommitlog)
+                return null;
+            else
+            {
+                Set<TableId> ids = new HashSet<>();
+                for (TableId id : mutation.getTableIds())
+                {
+                    if (keyspace.getColumnFamilyStore(id).writesShouldSkipCommitLog())
+                        ids.add(id);
+                }
+                mutation = mutation.without(ids);
+            }
+        }
+        // Note: It may be a good idea to precalculate none/all for the set of all tables in the keyspace,
+        // or memoize the mutation.getTableIds()->ids map (needs invalidation on schema version change).
```

Comment:
I'd vote for "memoize the mutation.getTableIds()->ids map" if invalidation isn't too hairy, given how hot this path is. You could rewrite this method in just a few lines..

1.) Get the `ids` for tables that skip the commit log.
2.) If the `ids` that should skip are the same as `mutation.getTableIds()`, return `null`.
3.) Otherwise, `mutation = mutation.without(ids)` and add to the `CommitLog`.

Reply:
The other thing I guess is that `noneSkipCommitlog` is going to be true for the existing implementation anyway and we'll avoid the `HashSet` creations.